analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Binary.Arithmetic.cs

```diff
@@ -57,6 +59,82 @@ private static NumberConstraint CalculateMultiply(NumberConstraint left, NumberC
         return NumberConstraint.From(min, max);
     }
 
+    private static NumberConstraint CalculateRemainder(NumberConstraint left, NumberConstraint right)
+    {
+        if (right.Equals(NumberConstraint.Zero))
+        {
+            return null;
+        }
+        else if (right.Min == 0)
+        {
+            right = NumberConstraint.From(1, right.Max);
+        }
+        else if (right.Max == 0)
+        {
+            right = NumberConstraint.From(right.Min, -1);
+        }
+        return NumberConstraint.From(CalculateRemainderMin(left, right), CalculateRemainderMax(left, right));
+    }
+
+    private static BigInteger? CalculateRemainderMin(NumberConstraint left, NumberConstraint right)
+    {
+        if (left.IsPositive)
+        {
+            return left.Min >= AbsoluteMin(right) ? 0 : left.Min;
+        }
+        else
+        {
+            var minDerivedFromRight = -AbsoluteMax(right) + 1;
+            if (minDerivedFromRight is null)
+            {
+                return left.Min;
+            }
+            else
+            {
+                return left.Min is null ? minDerivedFromRight : BigInteger.Max(left.Min.Value, minDerivedFromRight.Value);
+            }
+        }
+    }
+
+    private static BigInteger? CalculateRemainderMax(NumberConstraint left, NumberConstraint right)
+    {
+        if (left.IsNegative)
+        {
+            return left.Max <= -AbsoluteMin(right) ? 0 : left.Max;
+        }
+        else
+        {
+            var maxDerivedFromRight = AbsoluteMax(right) - 1;
+            if (maxDerivedFromRight is null)
+            {
+                return left.Max;
```

Comment:
Worth explaining that, when the range of the divisor is infinite, there will be always a divisor strictly bigger than any dividend, which would result into the dividend itself. Because the biggest dividend is positive (due to the fact that we are in the positive or mixed scenarios), then the biggest result is the max of the dividend range.